apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -312,11 +312,39 @@ def bind_port(self, context):
                       vnic_type)
             return
 
-        # Attempt to bind ports for DVS agents for nova-compute daemons
-        # first. This allows having network agents (dhcp, metadata)
-        # that typically run on a network node using an OpFlex agent to
-        # co-exist with nova-compute daemons for ESX, which host DVS agents.
         if context.current['device_owner'].startswith('compute:'):
+            # enforce the secure port binding rules if possible
+            if context.current['device_id']:
+                vm = nova_client.NovaClient().get_server(
+                    context.current['device_id'])
+                network_id = context.current.get('network_id')
+                net = self._get_plugin().get_network(context._plugin_context,
+                                                     network_id)
+                vrf = self._get_network_vrf(context, net)
+                aci_tenant = self.apic_manager.apic.fvTenant.name(
+                    vrf['aci_tenant'])
+                aci_vrf = self.apic_manager.apic.fvCtx.name(vrf['aci_name'])
+
+                ok_to_bind = True
+                secure_pb_list = self.net_cons.get_allow_vm_names(
```

Comment:
NetApp is asking this feature actually from APIC's point of view, that's why its implemented this way. Mandeep and Mike C. are well aware of this since the beginning and we actually even sent them a sample config to make sure this is what they want.